nemo/collections/nlp/models/language_modeling/megatron_gpt_peft_models.py

```diff
@@ -106,7 +106,7 @@ def load_state_dict(self, state_dict, strict: bool = True):
             assert set(state_dict.keys()) == self.adapter_keys
             super().load_state_dict(state_dict, strict=False)
         else:
-            super().load_state_dict(state_dict, strict=strict)
+            super().load_state_dict(state_dict, strict=True)
```

Comment:
maybe we should change this line back to `super().load_state_dict(state_dict, strict=strict)` to let the user decide

Reply:
This is done by the parent megatron model. I think we should only update speechllm code